tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -822,6 +823,13 @@ def permute_pads(pads):
 
     def _prelu_handler(self, trans, node):
         return self._handle_node_having_branches(trans, node)
+    
+    def _softmax_handler(self, trans, node):
+        # Softmax only operates on an arbitrary axis since opset 13
+        if self._g.opset >= 13:
+            return self._arg_min_max_handler(trans, node)
```

Comment:
Good to case on the opset.  I think all opsets should be able to be optimized, but the axis attribute has a different default value (1 in opsets 1 and 11, -1 in opsets 13 onward).

https://github.com/onnx/onnx/blob/master/docs/Changelog.md#Softmax-1

arg_min_max handler probably isn't the right thing to use.  Combine simple_through_handler with an update to the axis attribute.  The output shape of Softmax equals the input shape so you don't need the fancy shape updating from _arg_min_max_handler (inherited from _reduce_handler)

Reply:
Ah I see why you have the `self._g.opset >= 13` check.  The 2D coercion will be an issue if the transpose permutation crosses the coercion boundary.  If it doesn't though, you can handle pre-opset 13 cases by **not** updating the axis and using the simple through handler.